core/chains/evm/abi/selector_parser.go

```diff
@@ -0,0 +1,253 @@
+// Sourced from https://github.com/ethereum/go-ethereum/blob/fe91d476ba3e29316b6dc99b6efd4a571481d888/accounts/abi/selector_parser.go#L126
```

Comment:
I merged this in #12025 after fixing SonarQube complaints. Hopefully saved you some time but beware marge conflicts if you made any changes since then.